optimum/exporters/onnx/base.py

```diff
@@ -558,3 +558,95 @@ def get_decoder_onnx_config(
             f"{config.model_type} decoder export is not supported yet. ",
             f"If you want to support {config.model_type} please propose a PR or open up an issue.",
         )
+
+
+class OnnxConfigWithLoss(OnnxConfig, ABC):
+    """
+    Wrapper for the children classes of `optimum.exporters.onnx.OnnxConfig` to export the model through the ONNX format
+    with loss in outputs and labels in the inputs. For seq-to-seq models, labels will be appended to the inputs of
+    decoders.
+    """
+
+    _tasks_to_extra_inputs = {
+        "default": {"labels": {0: "batch_size"}},
+        "masked-lm": {"labels": {0: "batch_size", 1: "sequence_length"}},
+        "causal-lm": {"labels": {0: "batch_size", 1: "sequence_length"}},
+        "seq2seq-lm": {"labels": {0: "batch_size", 1: "sequence_length"}},
+        "sequence-classification": {"labels": {0: "batch_size"}},
+        "token-classification": {"labels": {0: "batch_size", 1: "sequence_length"}},
+        "multiple-choice": {"labels": {0: "batch_size"}},
+        "question-answering": {
+            "start_positions": {0: "batch_size"},
+            "end_positions": {0: "batch_size"},
+        },
+        "image-classification": {"labels": {0: "batch_size"}},
+        "seq2seq-lm": {"labels": {0: "batch_size", 1: "sequence_length"}},
+    }
+    _tasks_to_extra_outputs = {
+        "default": OrderedDict({"loss": {}}),
+    }
+
+    DUMMY_INPUT_GENERATOR_CLASSES = (DummyTrainingLabelsInputGenerator,)
+
+    def __init__(self, config: OnnxConfig):
+        self._onnx_config = config
+        self.task = self._onnx_config.task
+        self._normalized_config = self._onnx_config._normalized_config
+        self._patching_specs = self._onnx_config._patching_specs
+
+    @classmethod
+    def from_model_config(cls, config: OnnxConfig) -> "OnnxConfigWithLoss":
+        return cls(config)
+
+    @property
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        inputs = self._onnx_config.inputs
+        inputs.update(self._tasks_to_extra_inputs[self.task])
+        return inputs
+
+    @property
+    def outputs(self) -> Mapping[str, Mapping[int, str]]:
+        common_outputs = self._onnx_config.outputs
+        extra_outputs = self._tasks_to_extra_outputs["default"]
+        common_outputs.update(extra_outputs)
+        for key in reversed(extra_outputs.keys()):
+            common_outputs.move_to_end(key, last=False)
+        return copy.deepcopy(common_outputs)
+
+    def generate_dummy_inputs(self, framework: str = "pt", **kwargs):
+        dummy_inputs = self._onnx_config.generate_dummy_inputs(framework=framework, **kwargs)
+        input_name, _ = next(iter(self._onnx_config.inputs.items()))
```

Comment:
The way of getting the batch_size here seems a little bit hacky for me, but we might target it later I guess.